redis/src/main/scala/zio/redis/Output.scala

```diff
@@ -122,56 +114,58 @@ object Output {
   }
 
   case object StringOutput extends Output[String] {
-    protected def tryDecode(respValue: RespValue): String =
+    protected def tryDecode(respValue: RespValue)(implicit codec: Codec): String =
       respValue match {
         case RespValue.SimpleString(s) => s
         case other                     => throw ProtocolError(s"$other isn't a simple string")
       }
   }
 
   case object MultiStringOutput extends Output[String] {
-    protected def tryDecode(respValue: RespValue): String =
+    protected def tryDecode(respValue: RespValue)(implicit codec: Codec): String =
       respValue match {
         case s @ RespValue.BulkString(_) => s.asString
         case other                       => throw ProtocolError(s"$other isn't a bulk string")
       }
   }
 
   case object BulkStringOutput extends Output[Chunk[Byte]] {
-    protected def tryDecode(respValue: RespValue): Chunk[Byte] =
+    protected def tryDecode(respValue: RespValue)(implicit codec: Codec): Chunk[Byte] =
       respValue match {
         case RespValue.BulkString(value) => value
         case other                       => throw ProtocolError(s"$other isn't a bulk string")
       }
   }
 
+  final case class ArbitraryOutput[A: Schema]() extends Output[A] {
+    protected def tryDecode(respValue: RespValue)(implicit codec: Codec): A =
+      respValue match {
+        case RespValue.BulkString(s) => codec.decode(Schema[A])(s).fold(e => throw CodecError(e), identity)
```

Comment:
`getOrElse` is probably nicer than `fold` in this situation.

Reply:
Yeah, but the thing is that getOrElse doesn't give us a value in Left(v) :( I wanted to throw a specific error.